.github/workflows/publish.yml

```diff
@@ -0,0 +1,299 @@
+name: Publish Nargo
+
+on:
+  workflow_dispatch:
+    # Allow pushing a manual nightly release
+    inputs:
+      tag:
+        description: The tag to build Nargo from (leave empty to build a nightly release from master)
+        required: false
+      publish:
+        description: Whether to publish the build artifacts
+        type: boolean
+        default: false
+  schedule:
+    # Run a nightly release at 2 AM UTC
+    - cron: "0 2 * * *"
+  merge_group:
+  push:
+
+permissions:
+  # Necessary to upload new release artifacts
+  contents: write
```

Comment:
A future improvement we can make to this is to apply it only to a job and moving the release upload into its own job that relies on the various build jobs